src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,534 @@
+import Engine.Combat.BattleInstance.BattleParams;
+import Engine.GameEvents.GameEvent;
+import Engine.GameEvents.GameEventListener;
+import Engine.GameEvents.GameEventQueue;
+import Engine.GameEvents.MapChangeEvent;
+import Terrain.Environment;
+import Terrain.Environment.Weathers;
+import Terrain.GameMap;
+import Terrain.Location;
+import Terrain.MapMaster;
+import Terrain.TerrainType;
+import Units.Unit;
+import Units.UnitModel;
+
+/**
+ * Commander Ave (AH-vey) loves the cold, and her power allows her to inexorably, if slowly,
+ * grind her opponents down beneath a wall of ever-encroaching ice.
+ * Despite pernicious rumors to the contrary, she has nothing against Christmas.
+ * 
+ * Passive:
+ *    Ave generates snow around all owned properties, which spreads over time.
+ *    Her units take a movement and defense penalty in forests.
+ * 
+ * Glacio:
+ *    Boosts the snow-aura around her buildings;
+ *    Hail falls in a two-space radius around her units,
+ *      changing the weather in those tiles to snow,
+ *      damaging enemies for up to 2HP, and
+ *      destroying any forests (reducing them to grass).
+ *
+ * Likes: Sleigh rides, Hot Chocolate
+ * Dislikes: Jungles, Large Cats
+ */
+public class CommanderAve extends Commander
+{
+  public static final int SNOW_THRESHOLD = 100; // Big numbers for integer math.
+  public static final int SNOW_PER_TURN = 400;
+  public static final int SNOW_MELT_RATE = 100;
+  private int MAX_SNOW_SPREAD_RANGE = 5;
+  private CitySnowifier snowifier;
+
+  private static final CommanderInfo coInfo = new CommanderInfo("Ave", new instantiator());  
+  private static class instantiator implements COMaker
+  {
+    @Override
+    public Commander create()
+    {
+      return new CommanderAve();
+    }
+  }
+
+  private int[][] snowMap;
+
+  public CommanderAve()
+  {
+    super(coInfo);
+
+    // Ave's units are fine in the snow, but not in the trees.
+    for( UnitModel um : unitModels )
+    {
+      for( TerrainType terrain : TerrainType.TerrainTypeList )
+      {
+        um.propulsion.setMoveCost(Weathers.SNOW, terrain, um.propulsion.getMoveCost(Weathers.CLEAR, terrain));
+      }
+      for( Weathers weather : Weathers.values() )
+      {
+        um.propulsion.setMoveCost(weather, TerrainType.FOREST, um.propulsion.getMoveCost(weather, TerrainType.FOREST)+1);
+      }
+    }
+
+    addCommanderAbility(new GlacioAbility(this));
+    snowifier = new CitySnowifier(this);
+    GameEventListener.registerEventListener(snowifier);
+  }
+
+  @Override
+  public GameEventQueue initTurn(GameMap gameMap)
+  {
+    GameEventQueue returnEvents = super.initTurn(gameMap);
+
+    // Initialize our snow tracker if needed.
+    if( null == snowMap )
+    {
+      System.out.println("First time init: adding one snow");
+      // Start by initializing owned properties to SNOW_THRESHOLD
+      snowMap = new int[gameMap.mapWidth][gameMap.mapHeight];
+      addSnow(SNOW_THRESHOLD, gameMap, returnEvents);
```

Comment:
We add snow twice on turn one? Is this intended?

Reply:
If `null == snowMap`, yes. That means this is the first call to `initTurn`, so I add +100 to all of her properties so they have snow, and then fall through to the rest of the initialization routine where we get the +400 per property. This makes sure it spreads to the immediately-adjacent tiles right away. Same reason I have the `CitySnowifier` listener below - to add 100 on capture so the snow spreads as expected the next turn.

Reply:
Okay, but I feel like she doesn't really need to spread snow on turn 1.
It happens the next turn at the latest, and where multiple properties sit close, it is turn 1.